crates/matrix-sdk/src/room/mod.rs

```diff
@@ -750,6 +753,70 @@ impl Room {
         Ok(self.client.store().get_state_event_static_for_key(self.room_id(), state_key).await?)
     }
 
+    /// Returns the parents this room advertises as its parents.
+    ///
+    /// Results are in no particular order.
+    pub async fn parent_spaces(&self) -> Result<impl Stream<Item = ParentSpace> + '_> {
+        // Implements this algorithm:
+        // https://spec.matrix.org/v1.8/client-server-api/#mspaceparent-relationships
+
+        // Get all m.room.parent events for this room
+        Ok(self
+            .get_state_events_static::<SpaceParentEventContent>()
+            .await?
+            .into_iter()
+            // Extract state key (ie. the parent's id) and sender
+            .flat_map(|parent_event| match parent_event.deserialize() {
+                Ok(SyncOrStrippedState::Sync(SyncStateEvent::Original(e))) => {
+                    Some((e.state_key.to_owned(), e.sender))
+                }
+                Ok(SyncOrStrippedState::Sync(SyncStateEvent::Redacted(_))) => None,
+                Ok(SyncOrStrippedState::Stripped(e)) => Some((e.state_key.to_owned(), e.sender)),
+                Err(_) => None, // Ignore deserialization errors
+            })
+            // Check whether the parent recognizes this room as its child
+            .map(|(state_key, sender): (OwnedRoomId, OwnedUserId)| async move {
+                let Some(parent_room) = self.client.get_room(&state_key) else {
+                    // We are not in the room, cannot check if the relationship is reciprocal
+                    // TODO: try peeking into the room
+                    return ParentSpace::Unverifiable(state_key);
+                };
+                // Get the m.room.child state of the parent with this room's id
+                // as state key.
+                if let Ok(Some(child_event)) = parent_room
+                    .get_state_event_static_for_key::<SpaceChildEventContent, _>(self.room_id())
+                    .await
+                {
+                    if let Ok(SyncOrStrippedState::Sync(SyncStateEvent::Original(_))) =
+                        child_event.deserialize()
+                    {
```

Comment:
Both these `if let`s just swallow errors. For an I/O error in `get_state_event_static_for_key`, I guess it should be bubbled up.

A deserialization error on the event should probably just be logged (I'd suggest `info!`).